xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-api/src/main/java/org/xwiki/livedata/script/LiveDataScriptService.java

```diff
@@ -152,6 +161,64 @@ public String effectiveConfig(String liveDataConfigJSON)
         }
     }
 
+    /**
+     * Renders a Live Data as a {@link Block}.
+     *
+     * @param parameters the parameters to pass to the Live Data renderer
+     * @return the Live Data {@link Block}
+     * @throws LiveDataException in case of error when rendering the Live Data
+     * @since 16.0.0RC1
+     */
+    @Unstable
+    public Block render(Map<String, Object> parameters) throws LiveDataException
+    {
+        return render(parameters, null);
+    }
+
+    /**
+     * Renders a Live Data as a {@link Block}.
+     *
+     * @param parameters the parameters to pass to the Live Data renderer
+     * @param advancedParameters the advanced parameters to pass to the Live Data renderer
+     * @return the Live Data {@link Block}
+     * @throws LiveDataException in case of error when rendering the Live Data
+     * @since 16.0.0RC1
+     */
+    @Unstable
+    public Block render(Map<String, Object> parameters, Map<?, ?> advancedParameters) throws LiveDataException
+    {
+        return this.liveDataRenderer.render(convertParams(parameters), advancedParameters, false);
+    }
+
+    /**
+     * Renders a Live Data as a {@link String}.
+     * 
+     * @param parameters the parameters to pass to the Live Data executor
+     * @return the result of {@link #render(Map)} in the current syntax
+     * @throws LiveDataException in case of error when rendering the Live Data
+     * @since 16.0.0RC1
+     */
+    @Unstable
+    public String execute(Map<String, Object> parameters) throws LiveDataException
```

Comment:
It would be much more consistent with other API to do what I suggested:
* render -> string
* execute -> Block